sites/target.py

```diff
@@ -0,0 +1,199 @@
+from selenium import webdriver
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait as wait
+from webdriver_manager.chrome import ChromeDriverManager
+from chromedriver_py import binary_path as driver_path
+from utils import random_delay, send_webhook
+import string, random, settings, time, re
+
+class Target:
+    def __init__(self, task_id, status_signal, image_signal, product, profile, proxy, monitor_delay, error_delay):
+        self.task_id, self.status_signal, self.image_signal, self.product, self.profile, self.monitor_delay, self.error_delay = task_id, status_signal, image_signal, product, profile, float(
+            monitor_delay), float(error_delay)
+
+        starting_msg = "Starting Target"
+        self.browser = self.init_driver()
+        self.product_image = None
+
+        if settings.dont_buy:
+            starting_msg = "Starting Target in dev mode; will not actually checkout"
+
+        self.status_signal.emit(self.create_msg(starting_msg, "normal"))
+        self.status_signal.emit(self.create_msg("Logging In..", "normal"))
+        self.login()
+        self.monitor()
+        self.atc()
+        self.checkout()
+        self.submit_billing()
+
+        if not settings.dont_buy:
+            self.submit_order()
+        else:
+            self.status_signal.emit(self.create_msg("Mock Order Placed", "success"))
+            send_webhook("OP", "Target", self.profile["profile_name"], self.task_id, self.product_image)
+
+    # https://stackoverflow.com/questions/33225947/can-a-website-detect-when-you-are-using-selenium-with-chromedriver
+    def change_driver(self, loc):
+        print(loc)
+        fin = open(loc, 'rb')
+        data = fin.read()
+        val = "$cdc_" + "".join(random.choices(string.ascii_letters + string.digits, k=22)) + "_"
+
+        result = re.search(b"[$]cdc_[a-zA-Z0-9]{22}_", data)
+
+        if result is not None:
+            self.status_signal.emit(self.create_msg("Changing value in Chromedriver", "normal"))
+            data = data.replace(result.group(0), val.encode())
+            fin.close()
+            fin = open(loc, 'wb')
+            fin.truncate()
+            fin.write(data)
+            fin.close()
+        else:
+            fin.close()
+
+    def init_driver(self):
+        driver_manager = ChromeDriverManager()
+        driver_manager.install()
+        self.change_driver(driver_path)
+        browser = webdriver.Chrome(driver_path)
+
+        browser.execute_cdp_cmd("Page.addScriptToEvaluateOnNewDocument", {
+            "source": """
+                  Object.defineProperty(navigator, 'webdriver', {
+                   get: () => undefined
+                  })
+                """
+        })
+
+        return browser
+
+    def login(self):
+        self.browser.get("https://www.target.com")
+
+        self.browser.find_element_by_id("account").click()
```

Comment:
I would suggest adding a check to see if the user is already logged by looking for the "Sign in". That way we can save time if they're logged in already.